packages/medusa/src/repositories/product.ts

```diff
@@ -540,6 +553,25 @@ export class ProductRepository extends Repository<Product> {
     )
   }
 
+  /**
+   * Upserts shipping profile for products
+   * @param ids IDs of products to update
+   * @param shippingProfileId ID of shipping profile to assign to products
+   * @returns updated products
+   */
+  public async upsertShippingProfile(
+    ids: string[],
```

Comment:
nit: Make it explicit what we are upserting profiles with
```suggestion
    productIds: string[],
```